lib/flat_map/open_mapper.rb

```diff
@@ -55,13 +55,13 @@ def self.inherited(subclass)
     # @param [Object] target Target of mapping
     # @param [*Symbol] traits List of traits
     # @raise [FlatMap::Mapper::NoTargetError]
-    def initialize(target, *traits)
+    def initialize(target, *traits, &block)
       raise NoTargetError.new(self.class) unless target
 
       @target, @traits = target, traits
 
       if block_given?
-        singleton_class.trait :extension, &Proc.new
+        singleton_class.trait :extension, &block
```

Comment:
Interesting. I did not know that was possible in Ruby 2.x:
```ruby
def foo1(&b); b.call; end

def foo2; Proc.new.call; end

foo1 { p :hello }
 # => :hello

foo2 { p :hello }
 # => :hello
```

and then in Ruby 3.x:
```ruby
foo1 { p :hello }
 # => :hello

foo2 { p :hello }
 # (irb):2:in `new': tried to create Proc object without a block (ArgumentError)
 #         from (irb):2:in `foo2'
 #         from (irb):3:in `<main>'
 #         from /Users/peter.maneykowski/.rvm/rubies/ruby-3.1.3/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
 #         from /Users/peter.maneykowski/.rvm/rubies/ruby-3.1.3/bin/irb:25:in `load'
 #         from /Users/peter.maneykowski/.rvm/rubies/ruby-3.1.3/bin/irb:25:in `<main>'
 #         from /Users/peter.maneykowski/.rvm/gems/ruby-3.1.3/bin/ruby_executable_hooks:22:in `eval'
 #         from /Users/peter.maneykowski/.rvm/gems/ruby-3.1.3/bin/ruby_executable_hooks:22:in `<main>'
```